common/crypto/attestation-api/scripts/attestation_to_evidence.sh

```diff
@@ -0,0 +1,118 @@
+###########################################################
+function b64quote_to_iasresponse() {
+    #get api key
+    API_KEY_FILEPATH="${FPC_PATH}/config/ias/api_key.txt"
+    test -f ${API_KEY_FILEPATH} || die "no api key file ${API_KEY_FILEPATH}"
+    API_KEY=$(cat $API_KEY_FILEPATH)
+
+    #get verification report
+    QUOTE=$1
+    # contact IAS to get the verification report
+    IAS_RESPONSE=$(curl -s -H "Content-Type: application/json" -H "Ocp-Apim-Subscription-Key:$API_KEY" -X POST -d '{"isvEnclaveQuote":"'$QUOTE'"}' https://api.trustedservices.intel.com/sgx/dev/attestation/v4/report -i)
+    # check status (as there may be multiple header, we rather check presence of relevant fields)
+    echo "$IAS_RESPONSE" | grep "X-IASReport-Signature"           >/dev/null || die "IAS Response error"
+    echo "$IAS_RESPONSE" | grep "X-IASReport-Signing-Certificate" >/dev/null || die "IAS Response error"
+}
+
+###########################################################
+# iasresponse_to_evidence
+#   input:  ias response as parameter
+#   output: IAS_EVIDENCE variable
+###########################################################
+function iasresponse_to_evidence() {
+    IAS_RESPONSE="$1"
+    #encode relevant info in json format
+    IAS_SIGNATURE=$(echo "$IAS_RESPONSE" | grep -Po 'X-IASReport-Signature: \K[^ ]+' | tr -d '\r')
+    IAS_CERTIFICATES=$(echo "$IAS_RESPONSE" | grep -Po 'X-IASReport-Signing-Certificate: \K[^ ]+')
+    IAS_REPORT=$(echo "$IAS_RESPONSE" | grep -Po '{"id":[^ ]+')
+    JSON_IAS_RESPONSE=$(jq -c -n --arg sig "$IAS_SIGNATURE" --arg cer "$IAS_CERTIFICATES" --arg rep "$IAS_REPORT" '{iasSignature: $sig, iasCertificates: $cer, iasReport: $rep}')
+    #set output
+    IAS_EVIDENCE=$JSON_IAS_RESPONSE
+}
+
+###########################################################
+# simulated_to_evidence
+#   input:  evidence as parameter
+#   output: SIMULATED_EVIDENCE variable
+###########################################################
+function simulated_to_evidence() {
+    SIMULATED_EVIDENCE=$1
+}
+
+###########################################################
+# get_tag_make_variable
+#   input:  tag string (e.g, "TAG_X") as parameter
+#   output: tag string variable (e.g., TAG_X)
+###########################################################
+function get_tag_make_variable() {
+    TAGS_PATH="${FPC_PATH}/common/crypto/attestation-api/attestation/attestation_tags.h"
+    define_to_variable "$TAGS_PATH" "$1"
+}
+
+get_tag_make_variable "ATTESTATION_TYPE_TAG"
+get_tag_make_variable "ATTESTATION_TAG"
+get_tag_make_variable "EVIDENCE_TAG"
+get_tag_make_variable "SIMULATED_TYPE_TAG"
+get_tag_make_variable "EPID_LINKABLE_TYPE_TAG"
+get_tag_make_variable "EPID_UNLINKABLE_TYPE_TAG"
+
+###########################################################
+# attestation_to_evidence
+#   input:  attestation as parameter
+#   output: EVIDENCE variable
+###########################################################
+function attestation_to_evidence() {
```

Comment:
> I guess this script is the evidence <- AttestationToEvidence(attestation) from interfaces.attestation.md?

Yes.

> It is a bit hidden as one of the three API functions (all the others have separate directories with an intuitive name and a header file which defines exactly that interface).

Subjective. It fulfils the purpose according the design.
Anyway...

> This could be clarified by: calling the directory conversion, attestation_to_evidence or alike

done

> (and move the mrenclave script to test as this is only need there)

It's exactly because an application (for example the test) can use that that the script is where it is.

> make it clear from comments which is the public (e.g., in the header defining the exported function and its usage and label via comments internal functions as internal/private and/or split the file into two, one which is just the exported function and another one which are the utility functions.

In a file with ~100 lines and 5 functions, I'd rather say it is clear.
Anyway, divided even further.